substrate/client/consensus/babe/src/lib.rs

```diff
@@ -1597,6 +1619,23 @@ where
 				let prune_and_import = || {
 					prune_finalized(self.client.clone(), &mut epoch_changes)?;
 
+					if epoch_duration_changed {
+						epoch_changes
+							.import(
+								descendent_query(&*self.client),
+								parent_header.hash(),
+								*parent_header.number(),
+								*parent_header.parent_hash(),
+								current_epoch,
+							)
+							.map_err(|e| {
+								ConsensusError::ClientImport(format!(
+									"Error importing epoch changes: {}",
+									e
+								))
+							})?;
+					}
+
```

Comment:
https://github.com/paritytech/polkadot-sdk/blob/d1979d4be9c44d06c62aeea44d30f5f201d67b5e/substrate/client/consensus/babe/src/lib.rs#L1497

to make sure the first_in_epoch is true in the next epoch, we still need the current epoch in the `epoch_changes`